pytorch_lightning/utilities/cli.py

```diff
@@ -34,6 +41,80 @@
     ArgumentParser = object
 
 
+class _Registry(dict):
+    def __call__(
+        self,
+        cls: Optional[Type] = None,
+        key: Optional[str] = None,
+        override: bool = False,
+    ) -> "Optional[Type]":
+        """
+        Registers a class mapped to a name.
+
+        Args:
+            cls: the class to be mapped.
+            key: the name that identifies the provided class.
+        """
+        if key is None:
+            key = cls.__name__
```

Comment:
This would seem to indicate that `cls` can't be optional because it would fails if `key` not given.

Reply:
Done !